### PR TITLE
fix(net): double callback in net :_write

### LINKED ISSUE #697: Socket:_write() executes the callback twice in case of error
Causes uncaught exception:

```
Uncaught exception:
[string "../stream/stream_wr..."]:349: attempt to call local 'cb' (a nil value)
stack traceback:
	[string "../stream/stream_wr..."]:349: in function 'onwriteError'
	[string "../stream/stream_wr..."]:371: in function 'onwrite'
	[string "../stream/stream_wr..."]:137: in function 'callback'
	[string "../net.lua"]:97: in function <[string "../net.lua"]:94>
```

net.lua
```lua
function Socket:_write(data, encoding, callback)
  if not self._handle then return end
  timer.active(self)
  uv.write(self._handle, data, function(err)
    timer.active(self)
    if err then
      callback(err) -- <- second execution when we got an error
      self:destroy(err)
      return
    end
  end)
  callback() -- <- always executed
end
```

### PR DESCRIPTION
**Pull request by rphillips**
fixes #697

### CODE REVIEW

**Comment by creationix** [comment]
+1

File: tests/test-http-stream-finish.lua [hunk 1]
```diff
@@ -5,10 +5,6 @@ require('tap')(function(test)
     local server
     server = http.createServer(function (req, res)
       local body = "Hello world\n"
-      res:on("finish", expect(function()
```
Comment (rjemanuele) [inline tests/test-http-stream-finish.lua]: Removing this, removes the test for "finish"

File: tests/test-http-post-1mb.lua [hunk 2]
```diff
@@ -42,9 +42,9 @@ require('tap')(function(test)
       request:on('end', expect(function()
         p('server:onEnd')
         assert(postBuffer == data)
-        response:write(data)
```
Comment (rjemanuele) [inline tests/test-http-post-1mb.lua]: Moving server:close() seems ok but we shouldn't need to wrap the response:finish() in the callback.

File: deps/http.lua [hunk 3]
```diff
@@ -107,9 +108,9 @@ function ServerResponse:flushHeaders()
   self:writeHead(self.statusCode, self.headers)
 end
 
-function ServerResponse:write(chunk)
```
Comment (rjemanuele) [inline deps/http.lua]: nice

File: deps/http.lua [hunk 4]
```diff
@@ -122,8 +123,8 @@ function ServerResponse:finish(chunk)
   if #last > 0 then
     self.socket:write(last)
   end
-  self.socket:shutdown(function()
```
Comment (rjemanuele) [inline deps/http.lua]: what does the nextTick here fix?

File: tests/test-http-stream-finish.lua (hunk 1 above)
Comment (rphillips) [inline tests/test-http-stream-finish.lua]: fixed. put this back in.

**Comment by creationix** [comment]
+1

**Comment by rjemanuele** [comment]
+1

**Pull request merged**